e2e/pod.go

```diff
@@ -542,3 +553,72 @@ func validateRWOPPodCreation(
 
 	return nil
 }
+
+// verifyMountOpts verifies the mount options by editing the PV and
+// checking the mount options in nodeplugin container logs.
+func verifyMountOpts(
+	f *framework.Framework,
+	pvcPath, appPath, daemonSetName, cn, ns string,
+) error {
+	mountOption := "context=\"system_u:object_r:container_file_t:s0:c0,c1\""
+
+	// create PVC
+	pvc, err := loadPVC(pvcPath)
+	if err != nil {
+		framework.Failf("failed to load pvc: %v", err)
+	}
+	pvc.Namespace = f.UniqueName
+	err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+	if err != nil {
+		framework.Failf("failed to create PVC: %v", err)
+	}
+	// modify PV spec.MountOptions
+	pv, err := getBoundPV(f.ClientSet, pvc)
+	if err != nil {
+		framework.Failf("failed to get PV: %v", err)
+	}
+	pv.Spec.MountOptions = []string{mountOption}
+
+	// update PV
+	_, err = f.ClientSet.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update pv: %w", err)
+	}
+
+	app, err := loadApp(appPath)
+	if err != nil {
+		framework.Failf("failed to load application: %v", err)
+	}
+	app.Namespace = f.UniqueName
+	err = createApp(f.ClientSet, app, deployTimeout)
+	if err != nil {
+		framework.Failf("failed to create application: %v", err)
+	}
+
+	pod, err := f.ClientSet.CoreV1().Pods(f.UniqueName).Get(context.TODO(), app.Name, metav1.GetOptions{})
+	if err != nil {
+		framework.Logf("Error occurred getting pod %s in namespace %s", app.Name, f.UniqueName)
+
+		return fmt.Errorf("failed to get pod: %w", err)
+	}
+
+	nodepluginPodName, err := getDaemonsetPodOnNode(f, daemonSetName, pod.Spec.NodeName, ns)
+	if err != nil {
+		return fmt.Errorf("failed to get daemonset pod on node: %w", err)
+	}
+	logs, err := frameworkPod.GetPodLogs(context.TODO(), f.ClientSet, ns, nodepluginPodName, cn)
+	if err != nil {
+		return fmt.Errorf("failed to get pod logs from container %s/%s/%s : %w", ns, nodepluginPodName, cn, err)
+	}
+
+	if !strings.Contains(logs, mountOption) {
+		return fmt.Errorf("mount option %s not found in logs: %s", mountOption, logs)
+	}
```

Comment:
> verifying with logs doesn't ensure that the mount option is set, you need to inspect the mountpoint to check whether the mount option is applied on the mountpath or not.

@Madhu-1 do you have the cmd to do that ?

Reply:
you should be able to get it with `mount |grep <path>`.  see if you can reuse `checkMountOptions` function which we have already.

Reply:
Thanks!
 I'll re request review once ci is green with this change

Reply:
@Madhu-1  the selinux context is not shown in mount options but it is passed 
I guess that context is only visible on selinux enabled system. 
```
[38;5;9m[FAILED] failed to verify mount options: mount option context="system_u:object_r:container_file_t:s0:c0,c1" not found in 10.102.50.233:6789:/volumes/e2e/csi-vol-761f01c7-565d-4674-a154-809ba79de1c2/6551c48e-aff1-4e84-8b0f-a908984ac37f on /var/lib/www type ceph (rw,relatime,seclabel,name=cephcsi-cephfs-node,secret=<hidden>,fsid=00000000-0000-0000-0000-000000000000,acl,mds_namespace=myfs)
  [0m
```

https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e-helm_k8s-1.26/runs/369/nodes/100/log/?start=0 

ctrl+f: "context="
```
  I0614 10:22:00.902816   27952 cephcmds.go:105] ID: 21 Req-ID: 0001-0024-dadd4530-ed10-4771-8fcf-b55843578b77-0000000000000001-761f01c7-565d-4674-a154-809ba79de1c2 command succeeded: mount [-t ceph rook-ceph-mon-a.rook-ceph.svc.cluster.local:6789:/volumes/e2e/csi-vol-761f01c7-565d-4674-a154-809ba79de1c2/6551c48e-aff1-4e84-8b0f-a908984ac37f /var/lib/kubelet/plugins/kubernetes.io/csi/cephfs.csi.ceph.com/f400b7bc4e6e1ae0d881d5d12e9ff8a70cb54e6f642e6c2a79e3fb4b3ac58050/globalmount -o name=cephcsi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-111413977,mds_namespace=myfs,context="system_u:object_r:container_file_t:s0:c0,c1",_netdev]
```


I've reverted back to just checking the logs. 
The selinux context label is unique and since the mount is successful, I think it should be sufficient for now.

Reply:
something like

```
$ findmnt --json --output VFS-OPTIONS --source ...
```

Reply:
ok, so minikube does not use SELinux :disappointed: